### ISSUE TITLE
LSTM multidimension input

### ISSUE THREAD

**Issue by rkmp** [fchollet/keras#5608]
my input time series data is of the shape (nb_samples, 75, 32). 
75 is the timesteps and 32 is the input dimension.

```
model = Sequential()
model.add(LSTM(4, input_shape=(75, 32)))
model.summary()
```

The LSTM weight vectors,` [W_i, W_c, W_f, W_o]` are all 32 dimensions, but the output is just a single value. the output shape of the above model is (1,4). But in LSTM the output is also a vector so should not it be (32,4) for many to one implementation as above? why is it giving a single value for multi-dimension input also?

**Comment by unrealwill** [comment]
Hello,

Maybe you want to use `return_sequences=True ` in which case the output would be of shape (75,4)
See https://keras.io/layers/recurrent/#recurrent for more info.

In your (1,4) shape 1 refers to the numbers of samples, and 4 to the number of cells.
Model.summary has output_shape : (None, 4)   

I don't know how you could have (32,4)

**Comment by rajathkumarmp** [comment]
@unrealwill No that would be many to many implementation.

This is similar to what I am looking for,

<img width="465" alt="screen shot 2017-03-05 at 9 06 51 pm" src="https://cloud.githubusercontent.com/assets/6078431/23590028/b8d28696-01fe-11e7-97c1-0532a3e42942.png">

Here the input dimension is 4 and thus the output is a vector with 4 dimension.
so in my case the input dimension is 32 but in keras LSTM  I am getting just a single value, how do I get the complete 32 dim vector?

**Comment by rkmp** [comment]
No that would be many to many implementation.

This is similar to what I am looking for,

<img width="465" alt="screen shot 2017-03-05 at 9 06 51 pm" src="https://cloud.githubusercontent.com/assets/6078431/23590028/b8d28696-01fe-11e7-97c1-0532a3e42942.png">

Here the input dimension is 4 and thus the output is a vector with 4 dimension.
so in my case the input dimension is 32 but in keras LSTM  I am getting just a single value, how do I get the complete 32 dim vector?

**Comment by unrealwill** [comment]
I don't know what your mileage is but it seems to me that you misunderstood the structure of LSTM. (Maybe try with SimpleRNN, then GRU which have similar interfaces but are easier to comprehend).

If you want h to be of dim 32 then use LSTM(output_dim=32).

You can't easily access the hidden states of recurrent models in keras. 

Alternatively you can stack a Dense(32) on top of the LSTM so that you get a 32 dimensional vector.
